8-better-way-utilizing-object-pool-nocanvas-svg/js/index.js

```diff
@@ -27,22 +28,27 @@ Bubble.prototype = {
     this.vx = vx;
     this.vy = vy;
   },
-  update: function () {
-    this.x += this.vx;
-    this.y += this.vy;
-    this.vy += gravity;
-    this.vx = this.vx*friction;
-    this.vy = this.vy*friction;
+  update: function (dt) {
+    // friction opposes the direction of velocity
+    var acceleration = -Math.sign(this.vx) * friction;
+    // distance = velocity * time + 0.5 * acceleration * (time ^ 2)
+    this.x += this.vx * dt + 0.5 * acceleration * (dt ^ 2);
+    this.y += this.vy * dt + 0.5 * gravity * (dt ^ 2);
+    // velocity = velocity + acceleration * time
+    this.vy += gravity * dt;
+    this.vx += acceleration * dt;
```

Comment:
@sdras it should take more CPU, but its not the bottleneck. If you add more and more bubbles you'll probably notice that updating the DOM and painting it will be considerably slower, while the animation will still run smooth. This is because `requestAnimationFrame` will happen less frequent, however the offset will be still properly calculated based on the time difference between frames.

The old code didn't take the time into account and considered the constant interval between frames. So in case of long frames the animation would be slower, and in case of various frames, it would not be smooth.